[Bug]:[chaos][proxy]Proxy pods keep crashbackoff after running `testcases/test_e2e.py` or deleting pod failure chaos

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20211223-5a4dc83
- Deployment mode(standalone or cluster): cluster and proxy replicas is 2
- SDK version(e.g. pymilvus v2.0.0rc2): 2.0.0rc9.dev22
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

before chaos: one of the proxy pods will restart many times when running test_e2e.py
after chaos: affected pod by chaos can't recovery to ready status

### Expected Behavior

before chaos: all pod running well
after chaos: affected pod by chaos can recovery after deleting chaos

### Steps To Reproduce

```markdown
1. cd `tests/python_client/chaos`
2. change the tag in cluster-values.yaml
3. run `bash chaos_test.sh proxy pod_failure chaos-test 2`
```


### Anything else?

GitHub action: https://github.com/milvus-io/milvus/runs/4621347983?check_suite_focus=true
log: https://github.com/milvus-io/milvus/suites/4749870797/artifacts/131505942

local reproduce:
![image](https://user-images.githubusercontent.com/12268675/147330207-9998e308-f6e5-44c7-a8c9-24be8cdba7ed.png)
log:
[test-proxy-pod-failure-12-24-15-38-00.zip](https://github.com/milvus-io/milvus/files/7773252/test-proxy-pod-failure-12-24-15-38-00.zip)

error message 
```
time="2021-12-24T08:14:53Z" level=error msg="[Failed to create consumer]" consumerID=20 error="server error: ConsumerBusy: Exclusive consumer is already connected" name=wzekd subscription=by-dev-proxy-0 topic="persistent://public/default/by-dev-searchResult-0"
panic: Failed to create consumer by-dev-searchResult-0, error = All attempts results:
attempt #1:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #2:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #3:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #4:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #5:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #6:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #7:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #8:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #9:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #10:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #11:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #12:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #13:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #14:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #15:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #16:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #17:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #18:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #19:server error: ConsumerBusy: Exclusive consumer is already connected
attempt #20:server error: ConsumerBusy: Exclusive consumer is already connected


goroutine 264 [running]:
github.com/milvus-io/milvus/internal/msgstream.(*mqMsgStream).AsConsumerWithPosition(0xc0004d56c0, 0xc000c8e730, 0x1, 0x1, 0xc000603db0, 0xe, 0x0)
	/go/src/github.com/milvus-io/milvus/internal/msgstream/mq_msgstream.go:164 +0x2a7
github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).collectResultLoop(0xc000517680)
	/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:634 +0x102
created by github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).Start
	/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:845 +0x117
```

--- a/internal/datanode/data_node.go
+++ b/internal/datanode/data_node.go
@@ -209,11 +209,8 @@ func (node *DataNode) Init() error {
 		log.Error("DataNode init session failed", zap.Error(err))
 		return err
 	}
+	Params.DataNodeCfg.Refresh()
 
-	// TODO: caiyd need check
-	//Params.initMsgChannelSubName()
-	//TODO reset
-	//Params.initLogCfg()
 	log.Debug("DataNode Init",
 		zap.String("MsgChannelSubName", Params.DataNodeCfg.MsgChannelSubName),
 	)

--- a/internal/proxy/proxy.go
+++ b/internal/proxy/proxy.go
@@ -142,8 +142,7 @@ func (node *Proxy) Init() error {
 		log.Error("Proxy init session failed", zap.Error(err))
 		return err
 	}
-	// TODO: caiyd need check
-	//Params.initProxySubName()
+	Params.ProxyCfg.Refresh()
 	// wait for datacoord state changed to Healthy
 	if node.dataCoord != nil {
 		log.Debug("Proxy wait for DataCoord ready")

--- a/internal/querynode/query_node.go
+++ b/internal/querynode/query_node.go
@@ -186,6 +186,7 @@ func (node *QueryNode) Init() error {
 			initError = err
 			return
 		}
+		Params.QueryNodeCfg.Refresh()
 		connectEtcdFn := func() error {
 			etcdKV, err := etcdkv.NewEtcdKV(Params.QueryNodeCfg.EtcdEndpoints, Params.QueryNodeCfg.MetaRootPath)
 			if err != nil {
@@ -253,9 +254,6 @@ func (node *QueryNode) Init() error {
 			zap.Any("IP", Params.QueryNodeCfg.QueryNodeIP),
 			zap.Any("Port", Params.QueryNodeCfg.QueryNodePort),
 		)
-		// This param needs valid QueryNodeID
-		//Params.initMsgChannelSubName()
-		// TODO: caiyd need check
 	})
 
 	return initError

--- a/internal/util/paramtable/global_param.go
+++ b/internal/util/paramtable/global_param.go
@@ -541,6 +541,11 @@ func (p *proxyConfig) init(bp *BaseParamTable) {
 	//p.initRoleName()
 }
 
+// Refresh is called after session init
+func (p *proxyConfig) Refresh() {
+	p.initProxySubName()
+}
+
 // InitAlias initialize Alias member.
 func (p *proxyConfig) InitAlias(alias string) {
 	p.Alias = alias
@@ -1099,6 +1104,11 @@ func (p *queryNodeConfig) init(bp *BaseParamTable) {
 // InitAlias initializes an alias for the QueryNode role.
 func (p *queryNodeConfig) InitAlias(alias string) {
 	p.Alias = alias
+}
+
+// Refresh is called after session init
+func (p *queryNodeConfig) Refresh() {
+	p.initMsgChannelSubName()
 }
 
 func (p *queryNodeConfig) initCacheSize() {
@@ -1723,6 +1733,11 @@ func (p *dataNodeConfig) init(bp *BaseParamTable) {
 	p.initDeltaChannelName()
 
 	//p.initRoleName()
+}
+
+// Refresh is called after session init
+func (p *dataNodeConfig) Refresh() {
+	p.initMsgChannelSubName()
 }
 
 // InitAlias init this DataNode alias

